maps/src/main/java/org/odk/collect/maps/MapFragmentFactory.kt

```diff
@@ -1,7 +1,5 @@
 package org.odk.collect.maps
 
-import android.content.Context
-
 interface MapFragmentFactory {
-    fun createMapFragment(context: Context): MapFragment?
+    fun createMapFragment(): MapFragment
```

Comment:
We're now in an awkward situation of the result of this method having two expected types (`MapFragment` and `Fragment`). Ideally, `MapFragment` would be defined like this:

```kotlin
abstract class MapFragment : Fragment()
```

But, right now we can't do that because some implementations extend their mapping framework's Fragment. We might want to think about reworking them so that the mapping framework Fragment ends up as a child of the `MapFragment` instead, but I didn't want to add more rework to this PR.